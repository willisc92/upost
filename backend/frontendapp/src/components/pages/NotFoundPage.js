import { Link } from "react-router-dom";
import React from "react";

const NotFoundPage = () => (
    <div className="content-container">
        <h1>
            404! - <Link to="/">Go home</Link>
        </h1>
    </div>
);

export default NotFoundPage;
