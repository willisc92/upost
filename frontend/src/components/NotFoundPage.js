import { Link } from "react-router-dom";
import React from "react";

const NotFoundPage = () => (
    <div>
        404! - <Link to="/">Go home</Link>
    </div> // link works like an a-tag, but uses client-side routing.
);

export default NotFoundPage;
