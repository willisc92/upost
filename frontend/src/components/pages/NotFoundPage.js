import React from "react";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import ButtonBase from "@material-ui/core/ButtonBase";

const NotFoundPage = (props) => (
    <Box bgcolor="secondary.main" py={3}>
        <Container maxWidth="xl">
            <Typography variant="h1" color="error">
                404 - Page Not Found -{" "}
                <Typography variant="inherit">
                    <ButtonBase
                        onClick={() => {
                            props.history.push("/");
                        }}
                    >
                        Go home
                    </ButtonBase>
                </Typography>
            </Typography>
        </Container>
    </Box>
);

export default NotFoundPage;
