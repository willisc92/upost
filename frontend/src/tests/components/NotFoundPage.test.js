import React from "react";
import NotFoundPage from "../../components/pages/NotFoundPage";
import { shallow } from "enzyme";

test("Should render NotFoundPage component", () => {
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper).toMatchSnapshot();
});
