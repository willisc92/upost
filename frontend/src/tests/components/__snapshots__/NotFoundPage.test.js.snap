// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NotFoundPage component 1`] = `
<div
  className="content-container"
>
  <h1>
    404! - 
    <Link
      replace={false}
      to="/"
    >
      Go home
    </Link>
  </h1>
</div>
`;
